apstools/devices/motor_factory.py

```diff
@@ -0,0 +1,136 @@
+"""
+Motor Bundle Factory
+====================
+
+Factory for creating bundles with any number of positioners.
+
+.. autosummary::
+
+    ~axis_component
+    ~mb_class_factory
+    ~mb_creator
+
+.. rubric:: Examples
+
+.. code-block:: python
+    :linenos:
+
+    xy_stage = mb_creator(prefix="IOC:", motors={"x": "m21", "y": "m22"})
+
+.. seealso:: :doc:`How to use 'mb_creator()' </examples/ho_motor_factory>`
+"""
+
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Type
+from typing import Union
+
+from ophyd import Component
+from ophyd import EpicsMotor
+from ophyd import Kind
+from ophyd import MotorBundle
+from ophyd import SoftPositioner
+
+from apstools.utils import dynamic_import
+
+DEFAULT_DEVICE_BASE_CLASSES: List[Union[str, Type]] = [MotorBundle]
+DEFAULT_DEVICE_CLASS_NAME: str = "MB_Device"
+DEFAULT_MOTOR_LABELS: List[str] = ["motors"]
+DEFAULT_KIND: Kind = Kind.hinted | Kind.normal
+
+MOTORS_TYPE = Union[List[str], Dict[str, Optional[Union[str, Dict]]]]
+
+
+def axis_component(parms: dict[str, Any]) -> Component:
+    """Return a Component with 'parms for this axis."""
+    parms = parms or {}
+    if isinstance(parms, str):
+        parms = dict(prefix=parms)
+    attrs = dict(kind=DEFAULT_KIND, labels=DEFAULT_MOTOR_LABELS)
+    has_prefix: bool = parms.get("prefix") is not None
+    axis_class_name = parms.pop("class", None)
+    factory = parms.pop("factory", None)
+    if axis_class_name is not None and factory is not None:
+        raise ValueError(f"Unsupported configuration: {axis_class_name=!r} and {factory=!r}")
```

Comment:
Could this error message describe how the input parameters can be changed to make the configuration supported?

Reply:
Revised the exception message